### DISCUSSION TITLE
CNCF <> TODO OSPO BoF KubeCon Paris - Materials and Summary Notes

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#448] [OSPO Discussions]
This new discussion is designed for participants of the CNCF/TODO OSPO BoF at KubeCon Paris to share their summary notes in a thread 🧵 by the end of the session. It also includes the initial slide deck (in PDF format) that serves as the starting point for the BoF.

## Materials

[OSPOBoF_KubeConEU24](https://github.com/todogroup/ospology/files/14654353/OSPOBoF_Wed20_KubeConEU24.pdf)

## About OSPO BoF

Engage in discussions on OSPO day-to-day operations challenges when managing cloud-native technology and capabilities in the organization. Participants will have the opportunity to be part of the OSPO Book Project Contributors Group and meet like-minded professionals in the Cloud Native Computing Foundation (CNCF) Ecosystem

📆  Wednesday, March 20 • 16:00 - 18:00 (CET)
📍https://sched.co/1a9q9 (RSVP Required)

**Comment by ramrodo** [comment]
Group#1

Person 1: Between 50-70 projects in OSPO.

Person 2: CNCF Infrastructure lab, network, compute resources. Not traditional OSPO. 8 people OSS in part time. 

Person 3: Licenses, 200+ projects and figure out which are died. Project cross organizations.

Person 4: Not OSPO in the company but Contributor of Open Source.

Person 5: Finance Open Source. Different projects: internal, contribution, and maintain.

- Flow for contribution.
- We have a PlayBook, a guide of how to contribute. OpenSSF.
- Approval levels.
- Identify skills for the different needs of OSPO projects

10 years age of documentation. We outsourced the analysis.

Governance in this projects. 

Funeral for Open Source Project that are died. A ceremony for graving it. Event for funny.

Projects that are dependencies.

Track down old maintainers. Look for folks to maintain the project but ensure that they are using it.

We have a public repository and two our teams use it but not sure if other teams use it.

OSPO focuses on data analysis. There are tools for metrics on organization/repositories.

Identify the main contributors. Not all maintainers are kind.

Adopters file describe the use case. Maybe creating a GitHub discussion to ask who and how they are using the project.

If you leave an Open Source project, you did a good job.

**Best Practices**

- Recognition for CNCF to create an ecosystem where it didn't exist before. Keep in sync all the projects.
- Licenses first. Not every projects fits in a box. If a project doesn't generate traction... should we archive it?
- Some projects started with just one maintainer but then, other maintainers owner it.
- Archiving is ok but have documentation to back it up.
- Seek similar projects to get married.
- Star projects for fully supported and not supported. Then, hire someone to maintain.
- **Manage lifecycle of the repositories**
- Every quarter make a review all the projects and end repositories. Or maybe not be open source.

**Reply by ramrodo** [reply]
10 years age of documentation. We outsourced the analysis.

Governance in this projects. 

Funeral for Open Source Project that are died. A ceremony for graving it. Event for funny.

Projects that are dependencies.

Track down old maintainers. Look for folks to maintain the project but ensure that they are using it.

We have a public repository and two our teams use it but not sure if other teams use it.

OSPO focuses on data analysis. There are tools for metrics on organization/repositories.

Identify the main contributors. Not all maintainers are kind.

Adopters file describe the use case. Maybe creating a GitHub discussion to ask who and how they are using the project.

If you leave an Open Source project, you did a good job.

**Reply by ramrodo** [reply]
- Regular audit and maintenance of the projects. Archive them if it is needed.

**Comment by CsatariGergely** [comment]
Discussed points:
- Sustainable contributions
  - It is better to focus contributions to a narrow place most important for the business than doing unsustainable contributions to a wide area
- Standardisation on internal tooling to build products
  - Cloud native is a driver of this standardisation as the CNCF graduated projects are the de facto standard for application development
  - OSPO can take a role in this by defining of these
- How to avoid tragedy of the commons 
  - How to convince the business side of your company to have dedicated open source developers?
  - Find the person with the budget or engineering resources who would like to contribute
  - Find the right way to support your open source projects
- Options to run an OSPO with the capability to contribute
  - Have dedicated upstream developers in the OSPO
  - Do not have dedicated upstream developers in the OSPO
  - Have a budget of internal development assigned to the OSPO
  - Be able to pay developers
    - Use grants
- Recommendations
  - OSPOs should have budget to control developers or have developers
  - OSPOs can have active role in selecting used open source projects

Contributors of this discussion:
- Gergely Csatari (Nokia)
- Alessandro Lascu (iCubed srl)
- Anish Shanbhag (UC Berkeley)
- Matthias Meschede, @mmesch (Modus Create)
- Yann Hamdaoui, @yannham (Modus Create)

Things to read:
 - [Kara Sowles - The State of Funding Free & Open Source Software @ FOSDEM 2024](https://fosdem.org/2024/schedule/event/fosdem-2024-2751-the-state-of-funding-free-open-source-software/)
